Install failes on 'node-gyp rebuild'

I tried to install this package globally

MacOS High Sierra 10.13.6 (17G65)

Python 2.7.10
node: v.8.5.0
node-gyp: v.3.8.0
java -version
  java version "1.8.0_181"
  Java(TM) SE Runtime Environment (build 1.8.0_181-b13)
  Java HotSpot(TM) 64-Bit Server VM (build 25.181-b13, mixed mode)


```
npm i -g commit-msg
npm WARN deprecated minimatch@2.0.10: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue

> commit-msg@0.2.2 preuninstall /usr/local/lib/node_modules/commit-msg
> npm run hook-uninstall


> commit-msg@0.2.2 hook-uninstall /usr/local/lib/node_modules/commit-msg
> node lib/setup uninstall

Could not find git repo starting at /usr/local/lib/node_modules/commit-msg/lib
/usr/local/bin/commit-msg -> /usr/local/lib/node_modules/commit-msg/bin/validate
/usr/local/bin/git-commit-msg -> /usr/local/lib/node_modules/commit-msg/bin/git-commit-msg

> java@0.6.1 install /usr/local/lib/node_modules/commit-msg/node_modules/java
> node-gyp rebuild

  CXX(target) Release/obj.target/nodejavabridge_bindings/src/java.o
In file included from ../src/java.cpp:1:
In file included from ../src/java.h:9:
In file included from ../../nan/nan.h:182:
../../nan/nan_maybe_43_inl.h:221:17: warning: 'CloneElementAt' is deprecated [-Wdeprecated-declarations]
  return array->CloneElementAt(GetCurrentContext(), index);
                ^
/Users/[HOME]/.node-gyp/8.5.0/include/node/v8.h:3457:3: note: 'CloneElementAt' has been explicitly
      marked deprecated here
  V8_DEPRECATED("Cloning is not supported.",
  ^
/Users/[HOME]/.node-gyp/8.5.0/include/node/v8config.h:332:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
In file included from ../src/java.cpp:1:
In file included from ../src/java.h:9:
In file included from ../../nan/nan.h:188:
In file included from ../../nan/nan_new.h:189:
../../nan/nan_implementation_12_inl.h:40:29: warning: 'New' is deprecated [-Wdeprecated-declarations]
  return v8::BooleanObject::New(value).As<v8::BooleanObject>();
                            ^
/Users/[HOME]/.node-gyp/8.5.0/include/node/v8.h:4749:3: note: 'New' has been explicitly marked
      deprecated here
  V8_DEPRECATED("Pass an isolate", static Local<Value> New(bool value));
  ^
/Users/[HOME]/.node-gyp/8.5.0/include/node/v8config.h:332:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
In file included from ../src/java.cpp:1:
In file included from ../src/java.h:9:
../../nan/nan.h:590:20: error: no type named 'GCEpilogueCallback' in 'v8::Isolate'
      v8::Isolate::GCEpilogueCallback callback
      ~~~~~~~~~~~~~^
../../nan/nan.h:596:20: error: no type named 'GCEpilogueCallback' in 'v8::Isolate'
      v8::Isolate::GCEpilogueCallback callback) {
      ~~~~~~~~~~~~~^
../../nan/nan.h:601:20: error: no type named 'GCPrologueCallback' in 'v8::Isolate'
      v8::Isolate::GCPrologueCallback callback
      ~~~~~~~~~~~~~^
../../nan/nan.h:607:20: error: no type named 'GCPrologueCallback' in 'v8::Isolate'
      v8::Isolate::GCPrologueCallback callback) {
      ~~~~~~~~~~~~~^
../../nan/nan.h:1927:15: warning: 'SetAccessor' is deprecated [-Wdeprecated-declarations]
  return obj->SetAccessor(
              ^
/Users/[HOME]/.node-gyp/8.5.0/include/node/v8.h:3127:3: note: 'SetAccessor' has been explicitly marked
      deprecated here
  V8_DEPRECATED("Use maybe version",
  ^
/Users/[HOME]/.node-gyp/8.5.0/include/node/v8config.h:332:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
../src/java.cpp:897:40: warning: 'ToNumber' is deprecated [-Wdeprecated-declarations]
  v8::Local<v8::Number> val = info[0]->ToNumber();
                                       ^
/Users/[HOME]/.node-gyp/8.5.0/include/node/v8.h:2279:10: note: 'ToNumber' has been explicitly marked
      deprecated here
  inline V8_DEPRECATED("Use maybe version", Local<Number> ToNumber() const);
         ^
/Users/[HOME]/.node-gyp/8.5.0/include/node/v8config.h:332:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
../src/java.cpp:927:40: warning: 'ToNumber' is deprecated [-Wdeprecated-declarations]
  v8::Local<v8::Number> val = info[0]->ToNumber();
                                       ^
/Users/[HOME]/.node-gyp/8.5.0/include/node/v8.h:2279:10: note: 'ToNumber' has been explicitly marked
      deprecated here
  inline V8_DEPRECATED("Use maybe version", Local<Number> ToNumber() const);
         ^
/Users/[HOME]/.node-gyp/8.5.0/include/node/v8config.h:332:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
../src/java.cpp:956:40: warning: 'ToNumber' is deprecated [-Wdeprecated-declarations]
  v8::Local<v8::Number> val = info[0]->ToNumber();
                                       ^
/Users/[HOME]/.node-gyp/8.5.0/include/node/v8.h:2279:10: note: 'ToNumber' has been explicitly marked
      deprecated here
  inline V8_DEPRECATED("Use maybe version", Local<Number> ToNumber() const);
         ^
/Users/[HOME]/.node-gyp/8.5.0/include/node/v8config.h:332:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
../src/java.cpp:983:42: warning: 'ToNumber' is deprecated [-Wdeprecated-declarations]
    v8::Local<v8::Number> val = info[0]->ToNumber();
                                         ^
/Users/[HOME]/.node-gyp/8.5.0/include/node/v8.h:2279:10: note: 'ToNumber' has been explicitly marked
      deprecated here
  inline V8_DEPRECATED("Use maybe version", Local<Number> ToNumber() const);
         ^
/Users/[HOME]/.node-gyp/8.5.0/include/node/v8config.h:332:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
../src/java.cpp:1019:40: warning: 'ToNumber' is deprecated [-Wdeprecated-declarations]
  v8::Local<v8::Number> val = info[0]->ToNumber();
                                       ^
/Users/[HOME]/.node-gyp/8.5.0/include/node/v8.h:2279:10: note: 'ToNumber' has been explicitly marked
      deprecated here
  inline V8_DEPRECATED("Use maybe version", Local<Number> ToNumber() const);
         ^
/Users/[HOME]/.node-gyp/8.5.0/include/node/v8config.h:332:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
../src/java.cpp:1044:40: warning: 'ToNumber' is deprecated [-Wdeprecated-declarations]
  v8::Local<v8::Number> val = info[0]->ToNumber();
                                       ^
/Users/[HOME]/.node-gyp/8.5.0/include/node/v8.h:2279:10: note: 'ToNumber' has been explicitly marked
      deprecated here
  inline V8_DEPRECATED("Use maybe version", Local<Number> ToNumber() const);
         ^
/Users/[HOME]/.node-gyp/8.5.0/include/node/v8config.h:332:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
../src/java.cpp:1263:16: warning: 'TryCatch' is deprecated [-Wdeprecated-declarations]
  v8::TryCatch tryCatch;
               ^
/Users/[HOME]/.node-gyp/8.5.0/include/node/v8.h:8157:3: note: 'TryCatch' has been explicitly marked
      deprecated here
  V8_DEPRECATED("Use isolate version", TryCatch());
  ^
/Users/[HOME]/.node-gyp/8.5.0/include/node/v8config.h:332:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
10 warnings and 4 errors generated.
make: *** [Release/obj.target/nodejavabridge_bindings/src/java.o] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:262:23)
gyp ERR! stack     at emitTwo (events.js:125:13)
gyp ERR! stack     at ChildProcess.emit (events.js:213:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Darwin 17.7.0
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /usr/local/lib/node_modules/commit-msg/node_modules/java
gyp ERR! node -v v8.5.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok

> commit-msg@0.2.2 postinstall /usr/local/lib/node_modules/commit-msg
> npm run hook


> commit-msg@0.2.2 hook /usr/local/lib/node_modules/commit-msg
> node lib/setup

Could not find git repo starting at /usr/local/lib/node_modules/commit-msg/lib
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: java@0.6.1 (node_modules/commit-msg/node_modules/java):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: java@0.6.1 install: `node-gyp rebuild`
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Exit status 1
```

--- a/package.json
+++ b/package.json
@@ -49,6 +49,6 @@
     "semver": "^4.3.3"
   },
   "optionalDependencies": {
-    "java": "^0.6.1"
+    "java": "latest"
   }
 }

